lib/dry/schema/messages/yaml.rb

```diff
@@ -146,31 +141,49 @@ def prepare
 
       # @api private
       def pruned_data(template, **input)
-        tokens, = evaluation_context(template)
-        input.select { |k,| tokens.include?(k) }
+        evaluate_template(template) do |tokens:, **|
+          input.select { |k,| tokens.include?(k) }
+        end
       end
 
       # @api private
       def interpolate(template, **data)
-        _, evaluator = evaluation_context(template)
-        data.empty? ? evaluator.() : evaluator.(**data)
+        evaluate_template(template) do |evaluator:, **|
+          data.empty? ? evaluator.() : evaluator.(**data)
+        end
       end
 
       private
 
-      def evaluation_context(template)
-        cache.fetch_or_store(template.text) do
-          tokens = template.text.scan(TOKEN_REGEXP).flatten(1).map(&:to_sym).to_set
-          text = template.text.gsub('%', '#')
+      # @api private
+      def evaluate_template(template)
+        result = get(template.key, template.options)
+        return unless result
```

Comment:
@solnic do you think this should `raise`?

Reply:
Decided to raise a `KeyError` instead of returning `nil`, because this should only happen in exceptional cases